app/main/game.tsx

```
@@ -150,7 +147,31 @@ export default function GameScreen() {
     socket.on('gameOver', (p: any) => {
       if (!mounted) return;
       if (p?.game) dispatch(updateFromGameObject(p.game));
-      Alert.alert('Partie terminée', p.result || 'Fin');
+
+      let message = 'La partie est terminée.';
+      // Assurez-vous d'avoir l'ID de l'utilisateur actuel dans le store
+      const user = useAppSelector((state) => state.auth.user);
+
+      if (p.result === 'checkmate') {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)